### PR TITLE
Removed GeneralTags, ModelTags and DatasetTags

### PR DESCRIPTION
**Pull request by VictorHugoPilled**
Related to #1676. @Wauplin Thank you for the clearing up my confusion. Had to startover as some of my edits got mixed into a merge.

### CODE REVIEW

**Review by Wauplin — COMMENTED** [review]
What a very nice PR to witness! Thanks @VictorHugoPilled for removing all of this complex logic and making sure tests and docs are updated as well. I reviewed and made a few comments. The 2 main things to update IMO are:
- we should keep `get_model_tags` and `get_dataset_tags` as public methods. They are simple wrappers around an API endpoint. I would just update their return value to directly return a dict instead of AttributeDictionary.
- we should completely drop `AttributeDictionary` both in `huggingface_hub` and its tests

Thanks for working on that!

File: tests/test_endpoint_helpers.py [hunk 1]
```diff
@@ -14,17 +14,7 @@
...
-    DatasetTags,
-    GeneralTags,
-    ModelTags,
-)
-
-from .testing_utils import with_production_testing
+from huggingface_hub.utils.endpoint_helpers import AttributeDictionary
 
 
 class AttributeDictionaryCommonTest(unittest.TestCase):
```
[the remark this answers is not in the document]
Comment (VictorHugoPilled) [reply tests/test_endpoint_helpers.py:20]: @Wauplin Thank you for walking me through this process. I will work on these comments today.

**Comment by VictorHugoPilled** [comment]
@Wauplin All the changes you requested are complete.

**Review by julien-c — COMMENTED** [review]
yay!

**Review by Wauplin — APPROVED** [review]
Awesome! Thanks for making the changes @VictorHugoPilled :pray: 

image
^ love it! :smile:

**Comment by Wauplin** [comment]
Just merged this PR! Thanks for your contribution @VictorHugoPilled :fire:

**Pull request merged**